js/aofex.js

```diff
@@ -551,15 +551,13 @@ module.exports = class aofex extends Exchange {
         const side = this.safeString (trade, 'direction');
         const priceString = this.safeString (trade, 'price');
         const amountString = this.safeString2 (trade, 'amount', 'number');
-        const price = this.parseNumber (priceString);
-        const amount = this.parseNumber (amountString);
-        let cost = this.safeNumber (trade, 'total_price');
-        if (cost === undefined) {
-            cost = this.parseNumber (Precise.stringMul (priceString, amountString));
+        let costString = this.safeString (trade, 'total_price');
+        if (costString === undefined) {
```

Comment:
We should remove this conditional (lines 555-557), cause if the cost is not defined, it will be calculated in safeTrade.